monitor/core/src/main/java/core/ControllerImpl.java

```diff
@@ -0,0 +1,417 @@
+
+		String IPofMM = properties.getProperty( "IPofMM" );
+		String portOfMM= properties.getProperty( "portOfMM" );
+
+		if( IPofMM != null && portOfMM != null ) return MODACloudsMetrics.getRunningMetrics( IPofMM, portOfMM );
+
+
+		return null;
+	}
+
+	public String[] getAllMetrics(){
+
+		return MODACloudsMetrics.getAllMetrics();
+	}
+
+
+	public File getDataCollectors(){
+
+		return MODACloudsDataCollectors.getDataCollectorsJar( SERVER_DATA_COLLECTORS_PATH, DATA_COLLECTORS_FILE_NAME );
+	}
+
+	public File getDataCollector( String metricName ){
+
+		return MODACloudsDataCollectors.getDataCollectorJar( metricName, SERVER_DATA_COLLECTORS_PATH, DATA_COLLECTORS_FILE_NAME );
+	}
+
+	public String getDataCollectorInstallationFile( String metricName ){//It returns the content of the installation file of a data collector as a String.
+
+		Properties properties = readInitializationData( INITIALIZATION_CONFIGURATION_FILE_ON_SERVER );
+
+		String IPofKB = properties.getProperty( "IPofKB" );
+		String portOfKB= properties.getProperty( "portOfKB" );
+		String IPofDA= properties.getProperty( "IPofDA" );
+		String portOfDA= properties.getProperty( "portOfDA" );
+
+		if( IPofKB != null && portOfKB != null && IPofDA != null && portOfDA != null ) return MODACloudsDataCollectors.formInstallationExecutionCommand( metricName, IPofKB, portOfKB, IPofDA, portOfDA, DATA_COLLECTORS_FILE_NAME );
+
+
+		return null;
+	}
+
+
+	public String uninstallMonitoringRule( String id ){
+
+		String msg = "[INFO] Monitor controller: The monitoring rule has been deleted.";
+
+
+		Properties properties = readInitializationData( INITIALIZATION_CONFIGURATION_FILE_ON_SERVER );
+
+		String IPofMM = properties.getProperty( "IPofMM" );
+		String portOfMM= properties.getProperty( "portOfMM" );
+
+		if( IPofMM != null && portOfMM != null ) MODACloudsMonitoringRules.uninstallMonitoringRules( IPofMM, portOfMM, id );
+
+		else msg = "[ERROR] Monitor controller: you should firstly initialize MODACLouds monitoring platform.";
+
+
+		return msg;
+	}
+
+	public String addObserver( String metricName, String portOfObserver, String callbackURL ){
+
+		String msg = "[INFO] Monitor controller: The observer has been added.";
+
+
+		Properties properties = readInitializationData( INITIALIZATION_CONFIGURATION_FILE_ON_SERVER );
+
+		String IPofMM = properties.getProperty( "IPofMM" );
+		String portOfMM= properties.getProperty( "portOfMM" );
+
+		if( IPofMM != null && portOfMM != null ){
+
+			MODACloudsObservers.addObserver( IPofMM, portOfMM, metricName, callbackURL );
+
+			MODACloudsObservers.startObserver( portOfObserver );
+		}
+
+		else msg = "[ERROR] Monitor controller: you should firstly initialize MODACLouds monitoring platform.";
+
+
+		return msg;
+	}
+
+
+	//Main.
+	public static void main( String[] args ){
+
+		ControllerImpl controller = new ControllerImpl();
+
+		controller.initiateMonitor();
+
+
+		try{
+
+			System.out.println( "\nSeaClouds Monitoring Platform Menu:" );
+
+			while( true ){
```

Comment:
In case that you wanted to keep a similar CLI as Monitor.java, I'll recommend to uncoment necessary code and keep it separatedly.